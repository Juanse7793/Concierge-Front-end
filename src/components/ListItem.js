import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({
  name, start, end, deleteFunc,
}) => (
  <li className="row pill list">
    <p>{name}</p>
    <div className="row">
      <small>{`${start} ~ ${end}`}</small>
      <button type="button" className="red pill link" onClick={deleteFunc}>Delete</button>
    </div>
  </li>
);

ListItem.propTypes = {
  name: PropTypes.string.isRequired,
  start: PropTypes.string.isRequired,
  end: PropTypes.string.isRequired,
  deleteFunc: PropTypes.func.isRequired,
};

export default ListItem;
