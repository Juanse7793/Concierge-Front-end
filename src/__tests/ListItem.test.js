import renderer from 'react-test-renderer';
import ListItem from '../components/ListItem';

it('matches snapshot', () => {
  const component = renderer.create(
    <ListItem
      name="name"
      start="start"
      end="end"
      deleteFunc={() => {}}
      id={1}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
