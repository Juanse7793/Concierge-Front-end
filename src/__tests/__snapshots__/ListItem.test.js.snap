// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="row pill list"
>
  <p>
    name
  </p>
  <div
    className="row"
  >
    <small>
      start ~ end
    </small>
    <button
      className="red pill link"
      id={1}
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</li>
`;
